lib/Subscription/Manage.php

```diff
@@ -21,11 +22,42 @@ static function onSave() {
 
     if (!empty($subscriber_data['email']) && Subscriber::verifyToken($subscriber_data['email'], $token)) {
       if ($subscriber_data['email'] !== Pages::DEMO_EMAIL) {
-        $subscriber = Subscriber::createOrUpdate($subscriber_data);
-        $errors = $subscriber->getErrors();
+        $subscriber = Subscriber::createOrUpdate(static::filterOutEmptyMandatoryFields($subscriber_data));
+        $subscriber->getErrors();
       }
     }
 
     Url::redirectBack();
   }
+
+  private static function filterOutEmptyMandatoryFields(array $subscriber_data) {
+    $mandatory = self::getMandatory();
+    $filtered = [];
+    foreach ($subscriber_data as $name => $value) {
+      if (!isset($mandatory[$name]) || !empty(trim($value))) {
```

Comment:
if `$value` is `"0"` then this condition will be false and the field will be ignored!